php-checks/src/main/java/org/sonar/php/checks/UselessExpressionStatementCheck.java

```diff
@@ -77,13 +82,37 @@
     Kind.BOOLEAN_LITERAL
   };
 
+  private boolean fileContainsHTML;
+  private List<Tree> uselessNodes;
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    this.fileContainsHTML = false;
+    uselessNodes = new ArrayList<>();
+
+    super.visitCompilationUnit(tree);
+
+    if (!fileContainsHTML) {
+      for (Tree uselessNode : uselessNodes) {
+        context().newIssue(this, MESSAGE).tree(uselessNode);
+      }
+    }
+  }
+
   @Override
   public void visitExpressionStatement(ExpressionStatementTree tree) {
     ExpressionTree expression = tree.expression();
     if (expression.is(USELESS_KINDS)) {
-      context().newIssue(this, MESSAGE).tree(tree);
+      uselessNodes.add(tree);
     }
 
     super.visitExpressionStatement(tree);
   }
+
+  @Override
+  public void visitToken(SyntaxToken token) {
+    if (token.is(Kind.INLINE_HTML_TOKEN) && !token.text().equals("?>") && !token.text().equals("%>")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "?>" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1132)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "%>" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1132)